resteasy/src/main/java/com/linecorp/armeria/client/resteasy/ArmeriaResteasyClientBuilder.java

```diff
@@ -0,0 +1,541 @@
+        }
+
+        if (delegate.isTrustManagerDisabled()) {
+            clientFactoryBuilder.tlsNoVerify();
+        }
+        final KeyStore trustStore = delegate.getTrustStore();
+        final KeyStore keyStore = delegate.getKeyStore();
+        if (trustStore != null || keyStore != null) {
+            clientFactoryBuilder.tlsCustomizer(sslContextBuilder ->
+                                                       SslContextConfigurator.configureSslContext(
+                                                               sslContextBuilder,
+                                                               keyStore,
+                                                               delegate.getKeyStorePassword(),
+                                                               trustStore,
+                                                               delegate.isTrustSelfSignedCertificates())
+            );
+        }
+        webClientBuilder.factory(clientFactoryBuilder.build());
+        return webClientBuilder.build();
+    }
+
+    private ResteasyClient build(WebClient webClient) {
+        final long readTimeout = delegate.getReadTimeout(TimeUnit.MILLISECONDS);
+        final int bufferSize = delegate.getResponseBufferSize();
+        final ClientHttpEngine armeriaEngine = new ArmeriaJaxrsClientEngine(
+                webClient,  bufferSize > 0 ? bufferSize : ArmeriaJaxrsClientEngine.DEFAULT_BUFFER_SIZE,
+                readTimeout > 0 ? Duration.ofMillis(readTimeout) : null);
+        return delegate.httpEngine(armeriaEngine).build();
+    }
+
+    @Override
+    public ResteasyClient build() {
+        return build(buildWebClient());
+    }
+
+    /**
+     * {@link ArmeriaJaxrsClientEngine} will always be set as an {@link ClientHttpEngine}.
+     * @throws UnsupportedOperationException always
+     */
+    @Override
+    public ResteasyClientBuilder httpEngine(ClientHttpEngine httpEngine) {
+        // we'll always use an embedded ArmeriaJaxrsClientEngine
+        throw new UnsupportedOperationException("httpEngine");
+    }
+
+    @Override
+    public ClientHttpEngine getHttpEngine() {
+        return delegate.getHttpEngine();
+    }
+
+    @Override
+    public ResteasyClientBuilder useAsyncHttpEngine() {
+        // do nothing as ArmeriaJaxrsClientEngine is asynchronous
+        return delegate;
+    }
+
+    @Override
+    public boolean isUseAsyncHttpEngine() {
+        return true; // always return TRUE as ArmeriaJaxrsClientEngine is asynchronous
+    }
+
+    @Override
+    public ResteasyClientBuilder providerFactory(ResteasyProviderFactory providerFactory) {
+        delegate.providerFactory(providerFactory);
+        return delegate;
+    }
+
+    @Override
+    public ResteasyProviderFactory getProviderFactory() {
+        return delegate.getProviderFactory();
+    }
+
+    @Override
+    public ResteasyClientBuilder connectionTTL(long ttl, TimeUnit unit) {
+        delegate.connectionTTL(ttl, unit);
```

Comment:
How about setting directly to the `ClientFactoryBuilder` instead of using `delegate` not just for this method but for all methods?
I think we can throw `UnsupportedOperationException` if we can't set the parameter to `ClientFactoryBuilder`.

Reply:
Yes, it's certainly a possibility, but it would lead to much more complicated implementation. For instance, some of the values set are being processed by the delegate in a special way (e.g. Config).
So far, my strategy was to let the delegate do its job and pick up the results at the later stage during the `build()`.
I do throw `UnsupportedOperationException` in a few methods that I know for sure incompatible with Armeria API, like `SSLContext`.